workflows/host-genome-generation/host_genome_generation.wdl

```diff
@@ -1,150 +1,349 @@
+    input:
+    maybe_gz = genome_fasta_gz,
+    docker_image_id
+  }
+
+  call concatenate_and_unzip_fastas {
+    input:
+    fasta_gz = flatten([[genome_fasta.gz, ERCC_fasta_gz], other_fasta_gz]),
+    docker_image_id,
+  }
+
+  call bowtie2_build {
+    input:
+    fasta = concatenate_and_unzip_fastas.fasta,
+    genome_name,
+    docker_image_id,
+  }
+
+  call hisat2_build {
+    input:
+    fasta = concatenate_and_unzip_fastas.fasta,
+    transcripts_gtf_gz,
+    genome_name,
+    docker_image_id,
+  }
+
+  call kallisto_index {
+    input:
+    transcripts_fasta_gz = flatten([transcripts_fasta_gz, [ERCC_fasta_gz]]),
+    genome_name,
+    docker_image_id,
+  }
+
+  call minimap2_index as minimap2_index_dna {
+    input:
+    fasta = concatenate_and_unzip_fastas.fasta,
+    nucleotide_type = "dna",
+    genome_name,
+    docker_image_id,
+  }
+
+  call minimap2_index as minimap2_index_rna {
+    input:
+    fasta = concatenate_and_unzip_fastas.fasta,
+    nucleotide_type = "rna",
+    genome_name,
+    docker_image_id,
+  }
+
+  call star_generate {
+    input:
+    fasta = concatenate_and_unzip_fastas.fasta,
+    ERCC_fasta_gtf, 
+    transcripts_gtf_gz,
+    genome_name,
+    docker_image_id,
+  }
+
+  output {
+    File bowtie2_index_tar = bowtie2_build.index_tar
+    File hisat2_index_tar = hisat2_build.index_tar
+    File kallisto_idx = kallisto_index.idx
+    File minimap2_dna_mmi = minimap2_index_dna.index_mmi
+    File minimap2_rna_mmi = minimap2_index_rna.index_mmi
+    File star_genome_tar = star_generate.star_genome_tar
+
+    # also output the input files, to facilitate archival/provenance
+    File original_genome_fasta_gz = genome_fasta.gz
+    File? original_transcripts_gtf_gz = transcripts_gtf_gz
+    Array[File] original_transcripts_fasta_gz = transcripts_fasta_gz
+    File original_ERCC_fasta_gz = ERCC_fasta_gz
+    Array[File] original_other_fasta_gz = other_fasta_gz
+  }
+}
+
+task ensure_gz {
+  input {
+    File maybe_gz
+    String docker_image_id
+  }
+
+  String name = basename(maybe_gz)
+
+  command <<<
+    set -euxo pipefail
+    mkdir ans
+    if gzip -t '~{maybe_gz}'; then
+      cp '~{maybe_gz}' ans/
+    else
+      pigz -c -p 4 '~{maybe_gz}' > 'ans/~{name}.gz'
+    fi
+  >>>
+
+  output {
+      File gz = glob("ans/*")[0]
+  }
+
+  runtime {
+      docker: docker_image_id
+      cpu: 4
+      memory: "4GiB"
+  }
+}
+
+task concatenate_and_unzip_fastas {
+  input {
+    Array[File] fasta_gz
+    String docker_image_id
+  }
+
+  command <<<
+    pigz -dc ~{sep(' ',fasta_gz)} > "all.fasta"
+  >>>
+
+  output {
+      File fasta = "all.fasta"
+  }
+
+  runtime {
+      docker: docker_image_id
+      cpu: 4
+      memory: "4GiB"
+  }
+}
+
+task bowtie2_build {
+  input {
+    File fasta
+    String genome_name
+    Int seed = 42
+
+    Int cpu = 16
+    String docker_image_id
+  }
+
+  command <<<
+    set -euxo pipefail
+    TMPDIR=${TMPDIR:-/tmp}
+
+    mkdir -p "$TMPDIR"'/bt2/~{genome_name}'
```

Comment:
There's a discrepancy here with the old host filtering where the name of the folder of the old host filtering is 

"~{genome_name}.bowtie2"

This is causing inconsistency between the two and making it so that the new bowtie2 indices are not back compatible. We could create a symlink here before tarring to make it back-compatible. Otherwise we'd have to regenerate either the modern or old host filtering indices